modules/basic-skills/src/backend/callApi.ts

```diff
@@ -0,0 +1,47 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import { Transition } from './typings'
+
+const generateFlow = async (data: any, metadata: sdk.FlowGeneratorMetadata): Promise<sdk.FlowGenerationResult> => {
+  return {
+    transitions: createTransitions(data),
+    flow: {
+      nodes: createNodes(data),
+      catchAll: {
+        next: []
+      }
+    }
+  }
+}
+
+const createNodes = data => {
+  const nodes: sdk.SkillFlowNode[] = [
+    {
+      name: 'entry',
+      onEnter: [
+        {
+          type: sdk.NodeActionType.RunAction,
+          name: 'basic-skills/call_api',
+          args: {
+            url: data.url,
+            method: data.method,
+            body: data.body,
+            memory: data.memory,
+            variable: data.variable
+          }
+        }
+      ],
+      next: [{ condition: 'true', node: '#' }]
+    }
+  ]
+  return nodes
+}
+
+const createTransitions = (data): Transition[] => {
+  return [
+    { caption: 'On success', condition: `${data.memory}.${data.variable}`, node: '' },
+    { caption: 'On failure', condition: `!${data.memory}.${data.variable}`, node: '' }
+  ]
```

Comment:
How do we handle calls where nothing is returned? Maybe I just want to call an API to execute an action. Failure should means an axios error, while success/failure should be up to the user to specify (success = response 200, or has object x, or is not null, etc..)

Reply:
I think the skill should be as simple as possible. Its a quick way to call an api. If we add options to set headers and conditions to results in success or failure, we might as well create an action.

So the only thing that determines if the skill as failed is if axios throws an error. In that case, the error gets logged and the response is not set.

Reply:
yeah you're right, if I call an endpoint that returns nothing the exit condition will not work. I'll change that